src/script/lua_api/l_object.cpp

```diff
@@ -1803,6 +1803,11 @@ int ObjectRef::l_set_sky(lua_State *L)
 			// pop "sky_color" table
 			lua_pop(L, 1);
 		}
+		lua_getfield(L, 2, "fog");
+		if (lua_istable(L, -1)) {
+			sky_params.fog_distance = getfloatfield_default(L, -1,  "fog_distance", sky_params.fog_distance);
```

Comment:
`fog_distance` is a `s16`, this can overflow. Please clamp to s16 max and min.

Reply:
I would consider that user error, however this should be `getintfield_default`